src/table/column-filter.scss

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../menu/index';
+
+@mixin iui-column-filter {
+  @include iui-dropdown;
+  padding: $iui-baseline $iui-sm;
+  width: fit-content;
+
+  .iui-overlay {
+    position: relative;
+
+    > .iui-input {
+      padding-right: $iui-l * 2;
+    }
+
+    > .iui-borderless {
+      position: absolute;
```

Comment:
Can we avoid `absolute`?

Reply:
I don't know how else to position the calendar icon / borderless button right aligned within the text field.  I thought about exploring something like below, but seems like a bad idea:
```
<input>
  <button />
</input>
```

Reply:
Should be possible to do with flex but could get complicated 🤔

I guess it's fine for this component but when we add the dedicated combo input component, I would expect a proper solution.

Reply:
This can be achieved much simpler, I think:
![image](https://user-images.githubusercontent.com/36186912/116518005-65d82b00-a8d8-11eb-9cfb-cd5b9823cd31.png)
And also you need to revert changes in `labeled-inputs.scss`:
![image](https://user-images.githubusercontent.com/36186912/116518085-7dafaf00-a8d8-11eb-8d08-0dcdf85e91cc.png)
Result:
![image](https://user-images.githubusercontent.com/36186912/116517967-5bb62c80-a8d8-11eb-8485-ee7c816d4171.png)

Reply:
That's a good suggestion, I didn't think to try that.

However, I was trying to set this part up for a future component, which I think would need to handle the icon placement differently.  I have [a branch where I did some basic testing](https://github.com/iTwin/iTwinUI/compare/main...jon/input-combinations).  Below is a screenshot from that branch where I'm trying to throw every option at it.
![Screen Shot 2021-04-29 at 12 12 12 PM](https://user-images.githubusercontent.com/849817/116583727-81890480-a8cb-11eb-87c4-a72a6fcfda74.png)

Your solution would work in the column filter, but would not work if `.iui-inline` is not applied.

Reply:
If @mayank99 doesn't mind I will unresolve this conversation. If we want to achieve this combo with button and input then I think we don't need to use `position: absolute` but to have the same as button group. But then I don't think borderless button fits here as it is inside of the input field, not next to it as in your example in previous comment.
My recommendation would be maybe going with my example for now and when we implement combo input with button, we can refactor it. Because currently people are waiting for table filtering and some of them are not switching to itwinui from bwc because of this.

Reply:
We should be able to reuse inline input. We already have this covered, we just need to put a borderless icon instead of simple svg under `.iui-message`.
I have tried doing that in storybook (instead of svg icon, I passed IconButton) and got this one:
![image](https://user-images.githubusercontent.com/81580355/116660520-d1cf9780-a99b-11eb-907a-4dcb6023c99e.png)

So this is pretty close what we need, IMO. Just overriding `right` property value with 0 does the trick.
![image](https://user-images.githubusercontent.com/81580355/116660699-0fccbb80-a99c-11eb-8426-908497820f67.png)

From React side I think that would be pretty comfortable to do instead of having fully custom HTML.

Reply:
Followed the advice suggested here, I responded in [a different convo here](https://github.com/iTwin/iTwinUI/pull/51#discussion_r624160174).